pkg/log/locallogger/locallogger.go

```diff
@@ -1,24 +1,124 @@
 package locallogger
 
 import (
+	"crypto/sha256"
+	"encoding/json"
 	"fmt"
 	"io"
+	"sort"
+	"strings"
+	"sync"
+	"time"
 
 	"github.com/go-kit/kit/log"
 	"gopkg.in/natefinch/lumberjack.v2"
 )
 
 const (
 	truncatedFormatString = "%s[TRUNCATED]"
+
+	// Deduplication configuration
+	defaultCacheExpiry   = 5 * time.Minute // How long to remember log entries
+	defaultMaxCacheSize  = 1000            // Maximum number of unique log entries to track
+	cleanupInterval      = 1 * time.Minute // How often to clean up expired entries
+	duplicateLogInterval = 1 * time.Minute // How long to wait before logging a duplicate again
 )
 
+// Fields to exclude when creating content hash for deduplication
+var excludedHashFields = map[string]bool{
+	"ts":     true, // go-kit timestamp
+	"time":   true, // slog timestamp
+	"caller": true, // go-kit caller info
+	"source": true, // slog source info
```

Comment:
I would also add `original.time` and `original.source` here for desktop and watchdog logs